cpp/src/arrow/memory_pool.cc

```diff
@@ -534,4 +535,139 @@ int64_t ProxyMemoryPool::max_memory() const { return impl_->max_memory(); }
 
 std::string ProxyMemoryPool::backend_name() const { return impl_->backend_name(); }
 
+ReservationListener::~ReservationListener() {}
```

Comment:
this and the constructor aren't needed if you use = default as suggested in the header.

Reply:
done